src/main/java/core/basesyntax/FileWork.java

```diff
@@ -9,6 +14,15 @@
  */
 public class FileWork {
     public String[] readFromFile(String fileName) {
-        return null;
+        try {
+            return Files.lines(Path.of(fileName))
+                    .map(String::toLowerCase)
+                    .flatMap(line -> Stream.of(line.split("\\W")))
+                    .filter(word -> word.startsWith("w"))
+                    .sorted().toArray(String[]::new);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Don’t ignore exceptions. Re-throw them as RuntimeException(e.g. throw new RuntimeException(e)) if you don’t expect your code to fail here, or make necessary actions at the catch block to fix the problem.
https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_05.html